Ported/combat-bees/Assets/Scripts/Systems/MovementSystem.cs

```diff
@@ -22,22 +21,54 @@ public void Execute(in ArchetypeChunk chunk, int unfilteredChunkIndex, bool useE
         var transforms = chunk.GetNativeArray(TransformHandle);
         var velocities = chunk.GetNativeArray(VelocityHandle);
 
-        var gravityDt = TimeStep * Gravity; 
+        var gravityDt = TimeStep * Gravity;
         for (int i = 0; i < chunk.Count; ++i)
         {
             var tmComp = transforms[i];
             var velComp = velocities[i];
 
             var velNew = velComp.Value;
-            velNew.y += gravityDt; 
             var pos = tmComp.Value.Position;
+            float k_DeepImpactVel = 0.5f;
+            float k_Restitution = 0.4f;
+            float k_FrictionDamping = 0.3f;
+
+            velNew.y += gravityDt;
             pos += TimeStep * velNew;
-            // bounce off lower bottom
-            pos.y *= Convert.ToSingle(pos.y > 0);
+            float notOnGround = Convert.ToSingle(pos.y > 0);
+
+            if (notOnGround != 1.0f && velNew.y < -k_DeepImpactVel) // deep impact
+            {
+                velNew.y = -velNew.y * k_Restitution;
+                velNew.x *= k_FrictionDamping;
+                velNew.z *= k_FrictionDamping;
+            }
+            else
+            {
+                velNew.y *= notOnGround;
+                velNew.x *= notOnGround;
+                velNew.z *= notOnGround;
+            }
+
+            // stop at lower bottom
+            pos.y *= notOnGround;
 
             tmComp.Value.Position = pos;
             velComp.Value = velNew;
 
+            // derive heading of velocity
+            var heading = math.atan2(velNew.x, velNew.z);
+            var headingRot = Quaternion.AngleAxis(math.degrees(heading), Vector3.up);
+
+            // derive pitch of velocity, by bringing velocity into local space of object (with z forward) and
+            // calculating the angle of the velocity in the local yz plane of the object
+            var velForwardLocal = math.rotate(math.inverse(headingRot), velNew);
+            var pitch = math.atan2(velForwardLocal.y, velForwardLocal.z);
+            var pitchRot = Quaternion.AngleAxis(math.degrees(pitch), Vector3.left);
```

Comment:
Just FYI this is using UnityEngine Quaternion, and for consistent Unity.Mathematics usage you want to use lower case quaternion instead https://docs.unity3d.com/Packages/com.unity.mathematics@1.0/api/Unity.Mathematics.quaternion.html#Unity_Mathematics_quaternion_AxisAngle_Unity_Mathematics_float3_System_Single_

Reply:
Good one! I was looking for that yesterday and found that Quaternion function first 😅